src/kmeans.jl

```diff
@@ -390,4 +390,41 @@ function repick_unused_centers(X::AbstractMatrix{<:Real}, # in: the data matrix
         colwise!(ds, distance, v, X)
         tcosts = min(tcosts, ds)
     end
+
+"""
+get_cluster_assignments(X::Matrix{T}, R::KmeansResult; ...) -> Vector{Int}
+
+Perform the clustering assigment of ``n`` points into `k` clusters,
+using the learned prototopyes from the input `KmeansResult`. 
+
+Note: This method is usefull when clustering new data leveraging a fitted model.
+
+# Arguments
+- `X`: Input data to be clustered.
+- `R`: Fitted keamns result.
+"""
+function get_cluster_assignments(
+    X::Matrix{T}, 
+    R::KmeansResult, 
+    distance::SemiMetric=SqEuclidean()) where {F<:Function, T}
```

Comment:
I´ve moved it to utils with your signature, but I kept `X` as input array since other methods in the package use it for refering to a matrix of samples